core/chains/evm/config/chain_scoped_gas_estimator.go

```diff
@@ -118,6 +123,33 @@ func (g *gasEstimatorConfig) EstimateLimit() bool {
 	return *g.c.EstimateLimit
 }
 
+// ====== DAOracle ======
+type daOracleConfig struct {
+	c toml.DAOracle
+}
+
+func (d *daOracleConfig) OracleType() toml.OracleType {
+	return d.c.OracleType
+}
+
+// OracleAddress returns the address of the oracle contract and is only supported on the OP stack for now.
+func (d *daOracleConfig) OracleAddress() *types.EIP55Address {
+	if d.c.OracleType != toml.Optimism {
+		return nil
+	}
+	return d.c.OracleAddress
+}
+
+// CustomGasPriceCalldata returns the calldata for a custom gas price API and is only supported on the OP stack
+// for now.
+func (d *daOracleConfig) CustomGasPriceCalldata() string {
```

Comment:
this is outdated, `CustomGasPriceCalldata` will only be used by the `CustomCalldataOracle`

Reply:
updated this with a TODO and the custom calldata ticket number